shell.nix

```diff
@@ -0,0 +1,23 @@
+let
+  hsPkgs = import ./default.nix {};
+in
+  hsPkgs.shellFor {
+    # Include only the *local* packages of your project.
+    packages = ps: with ps; [
+      dear-imgui
+    ];
+
+    # Builds a Hoogle documentation index of all dependencies,
+    # and provides a "hoogle" command to search the index.
+    withHoogle = true;
+
+    # You might want some extra tools in the shell (optional).
+
+    # Some common tools can be added with the `tools` argument
+    tools = { cabal = "3.2.0.0"; hlint = "2.2.11"; };
+    # See overlays/tools.nix for more details
```

Comment:
This comment is a bit misleading as we don't have this file. I think it's referring to `haskell.nix/overlays/tools.nix`, but right now it's confusing - let's drop it:

```suggestion
```